confluent_kafka/avro/cached_schema_registry_client.py

```diff
@@ -83,6 +111,35 @@ def __exit__(self, *args):
     def close(self):
         self._session.close()
 
+    @staticmethod
+    def _configure_basic_auth(conf):
+        url = conf['url']
+        auth_provider = conf.get('basic.auth.credentials.source', 'URL').upper()
+        if auth_provider not in VALID_AUTH_PROVIDERS:
+            raise ValueError("basic.auth.credentials.source must be one of {}"
+                             .format(auth_provider, VALID_AUTH_PROVIDERS))
+
+        if auth_provider == 'SASL_INHERIT':
+            if conf.get('sasl.mechanisms', '').upper() == 'GSSAPI':
+                raise ValueError("SASL_INHERIT supports SASL mechanisms PLAIN and SCRAM only")
+            auth = (conf.get('sasl.username', None), conf.get('sasl.password'))
+        elif auth_provider == 'USERINFO':
```

Comment:
`SASL_INHERIT` has an underscore and `USERINFO` doesn't. in the java implementation, they both have an underscore.